### PR TITLE
Prime Injection Pulse impl.

### PR DESCRIPTION
**Pull request by andreika-git**
Not tested yet... WIP

### CODE REVIEW

File: firmware/hw_layer/backup_ram.h [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+#include "efiGpio.h"
+
+typedef enum {
+	/**
+	 * IAC Stepper motor position, 16-bit (stored in BKP0R 0..15)
+	 * Used in stepper.cpp
+	 */
+	BACKUP_STEPPER_POS,
+	/**
+	 * Ignition switch counter, 8-bit (stored in BKP0R 16..23)
```
Comment (rusefi) [inline firmware/hw_layer/backup_ram.h]: I do not understand the purpose of this data point. what is ignition switch?

File: firmware/controllers/trigger/main_trigger_callback.cpp [hunk 2]
```diff
@@ -512,6 +515,47 @@ void mainTriggerCallback(trigger_event_e ckpSignalType, uint32_t trgEventIndex D
...
+#if EFI_PROD_CODE || defined(__DOXYGEN__)
+	// First, we need a protection against 'fake' ignition switch on and off (i.e. no engine started), to avoid repeated prime pulses.
+	// So we check and update the ignition switch counter in non-volatile backup-RAM
+	uint32_t ignSwitchCounter = backupRamLoad(BACKUP_IGNITION_SWITCH_COUNTER);
+	// if we're just toying with the ignition switch, give it another chance eventually...
+	if (ignSwitchCounter > 10)
+		ignSwitchCounter = 0;
+	// start prime injection if this is a 'fresh start'
+	if (ignSwitchCounter == 0) {
+    	// fill-in the prime event struct
```
Comment (rusefi) [inline firmware/controllers/trigger/main_trigger_callback.cpp]: some evil code formatting here

File: firmware/hw_layer/backup_ram.h [hunk 3]
```diff
@@ -0,0 +1,33 @@
+/**
+ * @file	backup_ram.cpp
```
Comment (rusefi) [inline firmware/hw_layer/backup_ram.h]: would be amazing to extract  backup_ram refactoring into a separate PR. looks like refactoring is big enough to be on it's own

File: firmware/hw_layer/backup_ram.h (hunk 1 above)
Comment (andreika-git) [reply firmware/hw_layer/backup_ram.h]: https://en.wikipedia.org/wiki/Ignition_switch
The counter stores the number of times the ignition switch is turned on. 
See startPrimeInjectionPulse() in main_trigger_callback.cpp, it has comments on what's going on...

**Comment by andreika-git** [comment]
Now tested on a real cal! It works! And repetitive pulse protection too!

**Pull request merged**